### PR TITLE
 Add Ionic for JHipster blog post

### PR DESCRIPTION
**Pull request by mraible**
Example app at https://github.com/oktadeveloper/okta-ionic-jhipster-example. Please review README for issues as well.

### CODE REVIEW

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+## Develop a Blog App with OIDC Authentication
+
+To develop a blog application with JHipster, open a terminal window, create a directory, and run `jhipster`.
+
+```
+mkdir blog
+cd blog
+jhipster
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: This block renders a little plain. I'm not sure there is any way around it adding a `$` in front of each line doesn't play nice for the windows folks (and sometimes confuses the cut/pasting people)

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+| Use Hibernate 2nd level cache? | `Yes` |
+| Maven or Gradle? | `Gradle` |
+| Other technologies? | `<blank>` |
+| Client framework? | `Angular 5` |
+| Enable SASS support? | `No` |
+| Enable i18n? | `Yes` |
+| Native language of application? | `English` |
+| Additional languages? | `<blank>` |
+| Additional testing frameworks? | `<blank>` |
+| Install other generators? | `No` |
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Our table style is pretty basic here too. The rest of the docs have a blue table header and zebra stripping?

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        resource:
+            filter-order: 3
+            user-info-uri: https://{yourOktaDomain}.com/oauth2/default/v1/userinfo
+            token-info-uri: https://{yourOktaDomain}.com/oauth2/default/v1/introspect
+            prefer-token-info: false
+```
+
+On Okta, navigate to **Users** > **Groups**. Create `ROLE_ADMIN` and `ROLE_USER` groups and add your account to them.
+
+**TIP:** If you've installed e2e tests with Protractor, you'll need to modify them to use an Okta account when running integration tests. Change the default credentials in `src/test/javascript/e2e/account/account.spec.ts` and `src/test/javascript/e2e/admin/administration.spec.ts`.
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Cool tip on testing, but this implies to stick your user/password in test. Though, maybe folks will understand that and use a config/env var?

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+```
+cd ionic4j
+ionic serve
+```
+
+You'll see a screen with a sign-in button. Click on it, and you'll be redirected to Okta to authenticate.
+
+{% img blog/jhipster-ionic/ionic-okta-login.png alt:"Ionic Authenticated" width:"400" %}
+{% img blog/jhipster-ionic/ionic-welcome.png alt:"Ionic Welcome" width:"400" %}
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Change order of images
The welcome/sign in screenshot should be to the left of the widget

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+Now that you know authentication works, you can use the entity generator to generate Ionic pages for your data model. Run the following commands (in your `~ionic4j` directory) to generate screens for your entities.
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: missing `/` ->  `~/ionic4j` ?

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+Enter `../blog` as the path to your existing application. When prompted to regenerate entities and overwrite files, type "Y".
+
+The iOS emulator runs on port 8080, so you will need to change your backend to run on a different port when running in an emulator. Change port `8080` to `8888` in the following files:
+
+* `backend/src/main/resources/config/application-dev.yml`
+* `ionic4j/src/providers/api/api.ts`
+
+**NOTE:** You'll also need to add `http://localhost:8888/login` as a valid redirect URI in your Okta application.
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Maybe mention to do it in the Okta Developer Console? (lots different types of applications in this example).

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+{% img blog/jhipster-ionic/emulator-first-blog.png alt:"Emulator First Blog" width:"500" %}{: .center-image }
+
+You can try adding a second entry by clicking the + sign in the top right corner. If the keyboard doesn't show when you click on a field, navigate to **Hardware** > **Keyboard** > **Toggle Software Keyboard**.
+
+## Why Ionic instead of a PWA?
+
+I hope you've enjoyed this tour of Ionic for JHipster. Since JHipster supports progressive web apps (PWAs), you might be asking yourself, why Ionic instead of a PWA?
+
+The first version of 21-Points Health I wrote with JHipster 2.x was painful to use on a mobile device, mostly because it didn't have remember me support. The 4.x version of 21-Points Health is better, but it's still not great. If PWA support was better on the iPhone (and it [will be soon](https://twitter.com/rmondello/status/956256845311590400)!), I might think differently. In the meantime, I like that Ionic provides a UI that looks like a native app, and its animations and things like pull-to-refresh are hard to beat.
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Add link to 21-Points Health

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+I want to develop the **best user experience**. Native apps are painful to distribute, but they still seem to work better than PWAs (on iOS). The beauty of the way that Ionic integrates with JHipster is you can have both! You can turn your JHipster app into a PWA (instructions are in your app's README) *and* distribute a mobile app in the app store. If you slap on on some analytics, you see which one gets more usage, and determine the best client for yourself.
+
+## Learn More About Ionic and JHipster
+
+I've written a thing or two about JHipster and Ionic on this blog. If you'd like to see how to deploy your Ionic app to a mobile device, I recommend reading the [deploy to a mobile device](/blog/2017/05/17/develop-a-mobile-app-with-ionic-and-spring-boot#deploy-to-a-mobile-device) section of [Tutorial: Develop a Mobile App With Ionic and Spring Boot](/blog/2017/05/17/develop-a-mobile-app-with-ionic-and-spring-boot#deploy-to-a-mobile-device). I showed how to add OIDC authentication to an Ionic app in [Build an Ionic App with User Authentication](https://developer.okta.com/blog/2017/08/22/build-an-ionic-app-with-user-authentication).
```
Comment (bdemers) [inline _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Same link twice?  `/blog/2017/05/17/develop-a-mobile-app-with-ionic-and-spring-boot#deploy-to-a-mobile-device`

Comment (mraible) [reply _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: Good catch!

File: _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md (hunk 2 above)
Comment (bdemers) [reply _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: https://developer.okta.com/docs/api/resources/users#response-parameters
I don't think you need to worry about it for the context of this post/review

Comment (mraible) [reply _source/_posts/2018-01-30-jhipster-ionic-with-oidc-authentication.md]: I added an issue for us to do some research on this to see if it's a good idea: https://github.com/okta/okta.github.io/issues/1694

**Pull request merged**